exchanges/kraken/kraken_websocket.go

```diff
@@ -33,52 +35,67 @@ const (
 	krakenAuthWSURL          = "wss://ws-auth.kraken.com"
 	krakenWSSandboxURL       = "wss://sandbox.kraken.com"
 	krakenWSSupportedVersion = "1.4.0"
-	// WS endpoints
-	krakenWsHeartbeat            = "heartbeat"
-	krakenWsSystemStatus         = "systemStatus"
-	krakenWsSubscribe            = "subscribe"
-	krakenWsSubscriptionStatus   = "subscriptionStatus"
-	krakenWsUnsubscribe          = "unsubscribe"
-	krakenWsTicker               = "ticker"
-	krakenWsOHLC                 = "ohlc"
-	krakenWsTrade                = "trade"
-	krakenWsSpread               = "spread"
-	krakenWsOrderbook            = "book"
-	krakenWsOwnTrades            = "ownTrades"
-	krakenWsOpenOrders           = "openOrders"
-	krakenWsAddOrder             = "addOrder"
-	krakenWsCancelOrder          = "cancelOrder"
-	krakenWsCancelAll            = "cancelAll"
-	krakenWsAddOrderStatus       = "addOrderStatus"
-	krakenWsCancelOrderStatus    = "cancelOrderStatus"
-	krakenWsCancelAllOrderStatus = "cancelAllStatus"
-	krakenWsRateLimit            = 50
-	krakenWsPingDelay            = time.Second * 27
-	krakenWsOrderbookDepth       = 1000
+
+	// Websocket Channels
+	krakenWsHeartbeat               = "heartbeat"
+	krakenWsSystemStatus            = "systemStatus"
+	krakenWsSubscribe               = "subscribe"
+	krakenWsUnsubscribe             = "unsubscribe"
+	krakenWsSubscribed              = "subscribed"
+	krakenWsUnsubscribed            = "unsubscribed"
+	krakenWsSubscriptionStatus      = "subscriptionStatus"
+	krakenWsTicker                  = "ticker"
+	krakenWsOHLC                    = "ohlc"
+	krakenWsTrade                   = "trade"
+	krakenWsSpread                  = "spread"
+	krakenWsOrderbook               = "book"
+	krakenWsOwnTrades               = "ownTrades"
+	krakenWsOpenOrders              = "openOrders"
+	krakenWsAddOrder                = "addOrder"
+	krakenWsCancelOrder             = "cancelOrder"
+	krakenWsCancelAll               = "cancelAll"
+	krakenWsAddOrderStatus          = "addOrderStatus"
+	krakenWsCancelOrderStatus       = "cancelOrderStatus"
+	krakenWsCancelAllOrderStatus    = "cancelAllStatus"
+	krakenWsRateLimit               = 50
+	krakenWsPingDelay               = time.Second * 27
+	krakenWsOrderbookDefaultDepth   = 1000
+	krakenWsCandlesDefaultTimeframe = 1
 )
 
-// orderbookMutex Ensures if two entries arrive at once, only one can be
-// processed at a time
+var channelNames = map[string]string{
+	subscription.TickerChannel:    krakenWsTicker,
+	subscription.OrderbookChannel: krakenWsOrderbook,
+	subscription.CandlesChannel:   krakenWsOHLC,
+	subscription.AllTradesChannel: krakenWsTrade,
+	subscription.MyTradesChannel:  krakenWsOwnTrades,
+	subscription.MyOrdersChannel:  krakenWsOpenOrders,
+}
+var reverseChannelNames = map[string]string{}
+
+func init() {
+	for k, v := range channelNames {
+		reverseChannelNames[v] = k
```

Comment:
suggestion: can just populate in reverse in `channelNames` instead of having two maps if not resolve?

Reply:
Hard pass 😁